apps/06SpaceTravelToMars.js

```diff
@@ -10,6 +10,52 @@ var jobTypes = {
 };
 
 // Your code here
+function CrewMember(name, job, specialSkill) {
+  this.name = name;
+  this.job = job;
+  this.specialSkill = specialSkill;
+  this.ship = null;
+  this.enterShip = function(shipAssignment) {
+    this.ship = shipAssignment;
+    shipAssignment.crew.push(this);
+  }
+};
+
+function Ship(name, type, ability) {
+  this.name = name;
+  this.type = type;
+  this.ability = ability;
+  this.crew = [];
+  this.missionStatement = function() {
+    for (var i = 0; i < this.crew.length; i++) {
+      var currentCrew = this.crew[i].job;
+      if (jobTypes[currentCrew] === this.type) {
```

Comment:
Tiny bug in here the tests don't catch. This loop will ALWAYS exit the function (by returning) on the FIRST iteration of the loop (ok it doesn't when the loop doesn't run, but psh) We actually don't want the else block at all